docs/docs/dynamo-store/architecture.md

```diff
@@ -0,0 +1,117 @@
+| `p`artition | String                              | The stream name (hash key)                                     |
+| `i`ndex     | Number                              | Int32.MAX for `Tip` 0-N for calved batches (range key)         |
+| `b`ytes     | ?Number                             | Only exists in the Tip, number of bytes in predecessor batches |
+| `etag`      | ?String                             | Etag of latest update (used for OCC)                           |
+| `e`vents    | List(Map([Event](#event-schema)))   | [Events](#event-schema)                                        |
+| `c`ases     | List(String)                        | Event types                                                    |
+| `u`nfolds   | List(Map([Unfold](#unfold-schema))) | [Unfolds](#unfold-schema)                                      |
+| `a`ppends   | Number                              | The number of events appended in the latest update             |
+| `n`umber    | Number                              | The version of the stream                                      |
+
+<h2 id="event-schema">Event schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+| `x`actionId     | ?String | The correlation id of the event                              |
+| wh`y`           | ?String | The causation id (wh`y` did this event occur)                |
+
+<h2 id="unfold-schema">Unfold schema</h2>
+
+| Key             | Type    | Description                                                  |
+| --------------- | ------- | ------------------------------------------------------------ |
+| `i`ndex         | Number  | The index at which the unfold was generated                  |
+| `c`ase          | String  | The event type of the unfold                                 |
+| `t`imestamp     | String  | ISO8601 DateTime representing the time the event was written |
+| `D`ata encoding | ?Number | the encoding used for the `d`ata attribute                   |
+| `d`ata          | ?Binary | The event payload                                            |
+| `M`eta encoding | ?Number | the encoding used for the `m`eta attribute                   |
+| `m`eta          | ?Binary | The event metadata                                           |
+
+# Reactions
+
+Reactions are an often necessary part of event-sourced architectures. DynamoDB
```

Comment:
I'd try to flip this to convey that they are almost the point once you've gotten past the toy 101 samples - there's always people to notify, indexes to be maintained, or lifecycles to things humans care about. They're not a terrible thing you have to do. - they're a thing you _get_ to to decouple things